test-e2e/local-peers.js

```diff
@@ -0,0 +1,56 @@
+import { test } from 'brittle'
+import {
+  connectPeers,
+  createManagers,
+  disconnectPeers,
+  waitForPeers,
+} from './utils.js'
+
+test('Local peers discovery each other and share device info', async (t) => {
+  const mobileManagers = await createManagers(5, t, 'mobile')
+  const desktopManagers = await createManagers(5, t, 'desktop')
+  const managers = [...mobileManagers, ...desktopManagers]
+  connectPeers(managers, { discovery: true })
+  await waitForPeers(managers, { waitForDeviceInfo: true })
+  const deviceInfos = [
+    ...(await Promise.all(mobileManagers.map((m) => m.getDeviceInfo()))).map(
+      (deviceInfo) => ({ ...deviceInfo, deviceType: 'mobile' })
```

Comment:
There are currently two reasons for making these things async, would be good to discuss:
- We might move all SQLite queries into a separate worker thread (which would mean they must become async)
- In the future it would be nice to run these same tests on mobile, across an IPC channel that uses `rpc-reflector`, which makes all methods async, so by using async as default, we ensure that we can run the tests across an IPC channel.
The thought behind it is that changing sync code to async is much harder and requires more changes, and since in the app everything runs over IPC, everything is async anyway, so there is no performance gain from doing sync.